### PR TITLE
[stable/stolon] Make STKEEPER_PG_LISTEN_ADDRESS configurable

### PR DESCRIPTION
**Pull request by samene**
Signed-off-by: Sameer Mene <samene@gmail.com>

#### Is this a new chart
No

#### What this PR does / why we need it:
Currently there is no way to set the env variable `STKEEPER_PG_LISTEN_ADDRESS` in the keeper because this is always set in the command. The listen address needs to be set to "*" in certain scenarios, for example when istio sidecar injection is enabled. 

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes # 23896

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [ ] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/stolon/templates/keeper-statefulset.yaml [hunk 1]
```diff
@@ -52,7 +52,8 @@ spec:
               IFS='-' read -ra ADDR <<< "$(hostname)"
               export STKEEPER_UID="{{ .Values.keeper.uid_prefix }}${ADDR[-1]}"
               export POD_IP=$(hostname -i)
-              export STKEEPER_PG_LISTEN_ADDRESS=$POD_IP
+              export STKEEPER_PG_ADVERTISE_ADDRESS=$POD_IP
+              export STKEEPER_PG_LISTEN_ADDRESS=${STKEEPER_PG_LISTEN_ADDRESS:-$POD_IP}
```
Comment (lwolf) [inline stable/stolon/templates/keeper-statefulset.yaml:56]: I don't see a way to set this value through the chart.
Is it supposed to be set using `keeper.extraEnv` map?
if so, can we add some comment and/or example?

Comment (samene) [reply stable/stolon/templates/keeper-statefulset.yaml:56]: Yes, one would need to set it using ``keeper.extraEnv``. There is already a link in the chart [README](https://github.com/sorintlab/stolon/blob/master/doc/commands_invocation.md#commands-invocation) which lists all the command line options to keeper.
>Environment variables are prefixed with a string related to the command and take the name of the options but are UPPERCASE and any dashes are replaced by underscores.

**Pull request merged**